ledger/error/generator/lib/src/main/scala/com/daml/error/generator/ErrorCodeDocumentationGenerator.scala

```diff
@@ -8,239 +8,229 @@ import java.lang.reflect.Modifier
+        case otherAnnotationTypeName =>
+          throw new IllegalStateException(
+            s"Unexpected annotation of type: $otherAnnotationTypeName"
+          )
       }
-      .toSeq
-
-  private def convertToDocItem(error: ErrorCode): ErrorDocItem = {
-    val ErrorDocumentationAnnotations(deprecation, explanation, resolution) =
-      getErrorDocumentationAnnotations(error)
-
-    ErrorDocItem(
-      className = error.getClass.getName,
-      category = simpleClassName(error.category),
-      hierarchicalGrouping = error.parent,
-      conveyance = error.errorConveyanceDocString,
-      code = error.id,
-      deprecation = deprecation,
-      explanation = explanation,
-      resolution = resolution,
+    }
+    ErrorCategoryDocItem(
+      description = description.get,
+      resolution = resolution.get,
+      retryStrategy = retryStrategy.get,
     )
   }
 
-  private def convertToDocItem(group: ErrorGroup): GroupDocItem = {
-    val GroupDocumentationAnnotations(explanation) =
-      getGroupDocumentationAnnotations(group)
-
-    GroupDocItem(
-      errorClass = group.errorClass,
-      className = group.fullClassName,
-      explanation = explanation,
+  private def parseErrorCodeAnnotations(errorCode: ErrorCode): ErrorCodeAnnotations = {
+    val mirroredType = runtimeMirror.reflect(errorCode)
+    val annotations: Seq[ru.Annotation] = mirroredType.symbol.annotations
+    val deprecatedItem = new SettableOnce[DeprecatedItem]
+    val explanation = new SettableOnce[Explanation]
+    val resolution = new SettableOnce[Resolution]
+    annotations.foreach { annotation =>
+      getAnnotationTypeName(annotation) match {
+        case ExplanationTypeName =>
+          explanation.set(
+            Explanation(parseAnnotationValue(annotation.tree)),
+            context = ExplanationTypeName,
+          )
+        case ResolutionTypeName =>
+          resolution.set(
+            Resolution(parseAnnotationValue(annotation.tree)),
+            context = ResolutionTypeName,
+          )
+        case ScalaDeprecatedTypeName =>
+          deprecatedItem.set(
+            parseScalaDeprecatedAnnotation(annotation),
+            ScalaDeprecatedTypeName,
+          )
+        case otherAnnotationTypeName =>
+          throw new IllegalStateException(
+            s"Unexpected annotation of type: $otherAnnotationTypeName"
+          )
+      }
+    }
+    ErrorCodeAnnotations(
+      deprecation = deprecatedItem.get,
+      explanation = explanation.get,
+      resolution = resolution.get,
     )
   }
 
-  private def simpleClassName(any: Any): String =
-    any.getClass.getSimpleName.replace("$", "")
-
-  private case class ErrorDocumentationAnnotations(
-      deprecation: Option[DeprecatedDocs],
-      explanation: Option[Explanation],
-      resolution: Option[Resolution],
+  @nowarn(
+    "cat=unchecked&msg=abstract type pattern reflect.runtime.universe..* is unchecked since it is eliminated by erasure"
```

Comment:
Good call being very specific about the warning exclusion! 👍🏻 Sounds like something that can be avoided though. What causes this?

Reply:
I'm missing `import scala.reflect.runtime.universe._`. Fixing